airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/datetime_based_cursor.py

```diff
@@ -234,3 +233,12 @@ def _get_request_options(self, option_type: RequestOptionType, stream_slice: Str
         if self.end_time_option and self.end_time_option.inject_into == option_type:
             options[self.end_time_option.field_name] = stream_slice.get(self.partition_field_end.eval(self.config))
         return options
+
+    def should_be_synced(self, record: Record) -> bool:
+        record_cursor_value = record.get(self.cursor_field.eval(self.config))
+        if not record_cursor_value:
```

Comment:
It seems like it's problematic if there is no cursor value on a record, in the builder it looks like it works because you get your five pages, but in practice it will keep running.

Should we throw an exception in this case?

Reply:
That's a good question. I think I prefer to be fail safe and sync more records but I don't have data to back my intuition

Reply:
I don't think it's a big deal either way, we can also leave it as is

Reply:
Do we have a mechanism to do this? The only one I can see is the MessageRepository that does not support log messages for now (I think we decided to only allow of control messages but we can probably change that)

Reply:
Allowing logs from the MessageRepository is fine.

A lower lift approach would be to instantiate a [logger like we do for streams](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/core.py#L79). The obvious downside is we won't create AirbyteLogMessage from these, but they'll still show up as connector logs

Reply:
Added as part of this PR using the MessageRepository so that we can see it in the Connector Builder as well